src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+        lsls r0, 3\n\
+        adds r0, r1\n\
+        ldr r1, _0815B388 @ =TeachyTvRenderMsgAndSwitchClusterFuncs\n\
+        str r1, [r0]\n\
+    _0815B37A:\n\
+        pop {r4-r7}\n\
+        pop {r0}\n\
+        bx r0\n\
+        .align 2, 0\n\
+    _0815B380: .4byte gTeachyTV_StaticResources\n\
+    _0815B384: .4byte gTasks\n\
+    _0815B388: .4byte TeachyTvRenderMsgAndSwitchClusterFuncs\n\
+        ");
+}
+#endif
+
+void TeachyTvClusFuncTransitionRenderBg2TeachyTvGraphicInitNpcPos(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    TeachyTvBg2AnimController();
+    if ( ++data[2] > 63 )
+    {
+        CopyToBgTilemapBufferRect_ChangePalette(2, gUnknown_203F450->buffer4, 0, 0, 0x20, 0x20, 0x11);
+        TeachyTvSetSpriteCoordsAndSwitchFrame(((u8*)data)[2], 8, 0x38, 7);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+        PlayNewMapMusic(BGM_FRLG_FOLLOW_ME);
+    }
+}
+
+void TeachyTvClusFuncClearBg2TeachyTvGraphic(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if ( ++data[2] == 134 )
+    {
+        FillBgTilemapBufferRect_Palette0(2, 0, 2, 1, 0x1A, 0xC);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+    }
+}
+
+void TeachyTvClusFuncNpcMoveAndSetupTextPrinter(u8 taskId)
+{
+    s16 *data;
+    struct Sprite * spriteAddr;
+
+    data = gTasks[taskId].data;
+    spriteAddr = &gSprites[data[1]];
+    if ( data[2] != 35 )
+        ++data[2];
+    else {
+        if ( spriteAddr->pos2.x == 0x78 )
+        {
+            StartSpriteAnim(&gSprites[data[1]], 0);
+            TeachyTvInitTextPrinter(&gUnknown_841B83D);
+            data[2] = 0;
+            ++data[3];
+        }
+        else
+            ++spriteAddr->pos2.x;
+    }   
+}
+
+void TeachyTvClusFuncIdleIfTextPrinterIsActive(u8 taskId)
+{
+    u16* data = (u16 *)gTasks[taskId].data;
+    if ( !(sub_80BF518(0) << 16) )
+        ++data[3];
+}
+
+void TeachyTvRenderMsgAndSwitchClusterFuncs(u8 taskId)
+{
+    u16 *data;
+    data = gTasks[taskId].data;
+    if ( gMain.newKeys & 2 )
+    {
+        u8 *offset = &(gUnknown_203F450->var_4006);
+        u32 zero = 0;
+        offset[0] = 1;
+        TeachyTvSetSpriteCoordsAndSwitchFrame(*((char *)data + 2), 0, 0, 0);
+        FillWindowPixelBuffer(0, 0xCC);
+        CopyWindowToVram(0, 2);
+        TeachyTvClearBg1EndGraphicText();
+        data[2] = zero;
+        data[3] = zero;
+        gTasks[taskId].func = TeachyTvBackToOptionList;
+    }
+    else
+    {
+        void (***array)(u8) = &gUnknown_8479548;
+        void (**cluster)(u8) = array[(u8)gTeachyTV_StaticResources.optionChosen];
+        _call_via_r1(
+            taskId,
+            cluster[(s16)data[3]]);
+    }
+}
+
+void TeachyTvClusFuncTextPrinterSwitchStringByOptionChosen(u8 taskId)
+{
+    u16 *data = (u16 *)gTasks[taskId].data;
+    const char * const* texts = &gUnknown_8479560;
+    TeachyTvInitTextPrinter(texts[gTeachyTV_StaticResources.optionChosen]);
```

Comment:
Temporary variable `texts` is completely unnecessary.

Reply:
Actually if I inject it into next line the function won't match

Reply:
This might be a `static const` definition inside the function.